mongo/options/clientencryptionoptions.go

```diff
@@ -52,6 +58,14 @@ func (c *ClientEncryptionOptions) SetTLSConfig(tlsOpts map[string]*tls.Config) *
 	return c
 }
 
+// SetHTTPClient specifies the http.Client to be used for any KMS provider making HTTP requests.
+//
+// This should only be used to set custom HTTP client configurations. By default, the connection will use an internal.DefaultHTTPClient.
+func (c *ClientEncryptionOptions) SetHTTPClient(client *http.Client) *ClientEncryptionOptions {
```

Comment:
I think @matthewdale and I are still wondering what the use-case is for supplying a custom `http.Client` for KMS provider requests and OCSP requests. Is there a reason to have this level of granularity instead of just allowing setting `http.Client` on `ClientOptions` to be used for both KMS provider and OCSP requests? We're probably fine with it if so, just wondering if you had an example.

Reply:
So I did this because it looks like many of the examples in the codebase use these features more directly and adding the options ended up being easier. When not used, we do end up preferring the mongo client options anyway. If you think it makes more sense to not have the options and make sure they're stuffed into the config, I'm fine with that as well.

Reply:
I think I'd prefer removing these setters and adding a `SetHTTPClient` to `mongo.ClientOptions` if that sounds good to you.